cpp/src/join/hash_join.cu

```diff
@@ -47,34 +47,36 @@ namespace {
  * @brief Calculates the exact size of the join output produced when
  * joining two tables together.
  *
- * @throw cudf::logic_error if JoinKind is not INNER_JOIN or LEFT_JOIN
- *
- * @tparam JoinKind The type of join to be performed
+ * @throw cudf::logic_error if join is not INNER_JOIN or LEFT_JOIN
  *
  * @param build_table The right hand table
  * @param probe_table The left hand table
  * @param hash_table A hash table built on the build table that maps the index
  * of every row to the hash value of that row.
+ * @param join The type of join to be performed
  * @param nulls_equal Flag to denote nulls are equal or not.
  * @param stream CUDA stream used for device memory operations and kernel launches
  *
  * @return The exact size of the output of the join operation
  */
-template <join_kind JoinKind>
-std::size_t compute_join_output_size(table_device_view build_table,
-                                     table_device_view probe_table,
-                                     cudf::detail::multimap_type const& hash_table,
-                                     bool const has_nulls,
-                                     cudf::null_equality const nulls_equal,
-                                     rmm::cuda_stream_view stream)
+std::size_t compute_join_output_size(
+  table_view const& build_table,
+  table_view const& probe_table,
+  std::shared_ptr<cudf::experimental::row::equality::preprocessed_table> preprocessed_build,
+  std::shared_ptr<cudf::experimental::row::equality::preprocessed_table> preprocessed_probe,
+  cudf::detail::multimap_type const& hash_table,
+  join_kind join,
+  bool const has_nulls,
+  cudf::null_equality const nulls_equal,
```

Comment:
Please use `const` consistently. Either use for all, or none. Since most other places don't use `const` for plain type parameters, let's adopt that style here.